src/DataObjects/Document.php

```diff
@@ -375,10 +375,10 @@ public function getAllVersions(OperationContextInterface $context = null)
 
         $objectFactory = $this->getSession()->getObjectFactory();
         $result = array();
-        if (count($versions) !== null) {
+        if (count($versions)) {
             foreach ($versions as $objectData) {
                 $document = $objectFactory->convertObject($objectData, $context);
-                if (!(!$document instanceof DocumentInterface)) {
+                if (!($document instanceof DocumentInterface)) {
                     // this should never happen
```

Comment:
`convertObject` may return a non-Document and this clause guards against that. However, if such a response is encountered it is most definitely a (big) problem with the repository implementation - it is very much unexpected that the repository would return anything other than Documents when requesting versions of a Document which is the only resource that supports versions. Replaced comment with throwing of exception.